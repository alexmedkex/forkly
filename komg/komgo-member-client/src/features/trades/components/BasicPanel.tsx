import * as React from 'react'
import styled from 'styled-components'
import { media } from '../../../utils/media'

const Panel = styled.ul`
  list-style: none;
  margin: 0;
  ${(props: BasicPanelProps) => (props.padding ? `padding: ${props.padding}` : `padding: 30px 0px 30px 0px;`)};
  display: flex;
  flex-direction: column;
  ${(props: BasicPanelProps) => props.centeredForm && `margin: 0 auto; width: 800px;`};
  ${() => media.desktop`padding-left: 40px;`};
  ${(props: BasicPanelProps) => props.maxHeight && `max-height: ${props.maxHeight}px`};
`

interface BasicPanelProps {
  maxHeight?: number
  centeredForm?: boolean
  padding?: string
}

const BasicPanel: React.FC<BasicPanelProps> = ({ children, ...props }) => <Panel {...props}>{children}</Panel>

export default BasicPanel
