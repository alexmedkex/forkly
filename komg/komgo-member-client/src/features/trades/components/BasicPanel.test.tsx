import * as React from 'react'
import * as renderer from 'react-test-renderer'
import BasicPanel from './BasicPanel'

it('should match snapshot', () => {
  const tree = renderer.create(<BasicPanel>OK</BasicPanel>).toJSON()

  expect(tree).toMatchSnapshot()
})
