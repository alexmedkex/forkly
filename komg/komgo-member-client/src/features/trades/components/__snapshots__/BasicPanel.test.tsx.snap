// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  list-style: none;
  margin: 0;
  padding: 30px 0px 30px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:68.75em) {
  .c0 {
    padding-left: 40px;
  }
}

<ul
  className="c0"
>
  OK
</ul>
`;
